tree/ntuple/v7/src/RField.cxx

```diff
@@ -2368,6 +2394,50 @@ void ROOT::Experimental::RVariantField::CommitCluster()
 
 //------------------------------------------------------------------------------
 
+ROOT::Experimental::RSetField::RSetField(std::string_view fieldName, std::unique_ptr<Detail::RFieldBase> itemField)
+   : ROOT::Experimental::RProxiedCollectionField(fieldName, "std::set<" + itemField->GetType() + ">")
+{
+}
+
+std::unique_ptr<ROOT::Experimental::Detail::RFieldBase>
+ROOT::Experimental::RSetField::CloneImpl(std::string_view newName) const
+{
+   auto newItemField = fSubFields[0]->Clone(fSubFields[0]->GetName());
+   return std::make_unique<RSetField>(newName, std::move(newItemField));
+}
+
+void ROOT::Experimental::RSetField::ReadGlobalImpl(NTupleSize_t globalIndex, Detail::RFieldValue *value)
```

Comment:
I'm curious as to why we need to override `ReadGlobalImpl()` in `RSetField`.  Couldn't we just use the iterator-based implementation in `RProxiedCollectionField`?

Reply:
After #13197 is merged (hopefully still today), you should be able to rebase your PR atop and this member function should not require overriding anymore, i.e. we should in principle stick to the implementation in the base class.  :slightly_smiling_face: